specification/subscription/resource-manager/Microsoft.Subscription/preview/2018-11-01-preview/subscriptions.json

```diff
@@ -156,6 +156,70 @@
           }
         ]
       }
+    },
+    "/providers/Microsoft.Subscription/subscriptions/{subscriptionId}/cancel": {
+      "post": {
+        "operationId": "CancelSubscription",
+        "description": "Cancels the subscription",
+        "parameters": [
+          {
+            "name": "subscriptionId",
+            "in": "path",
+            "description": "Subscription Id.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/CanceledSubscriptionId"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Subscription/subscriptions/{subscriptionId}/rename": {
```

Comment:
How will rename work? Will it generate a new unique GUID? If yes, POST is more suited. What is the scenario in which customer may want to rename?
Patch will not work for this URL.

Reply:
Changed it to post

Reply:
please add request body and include displayName or whatever property they can update using this API